monitoring/bin/deploy_monitoring_openshift.sh

```diff
@@ -40,325 +40,324 @@ helmRepoAdd grafana https://grafana.github.io/helm-charts
-  -f "$autogeneratedYAMLFile" \
-  -f "$userGrafanaYAML" \
-  -f $tempoDSFile \
-  --set 'grafana\.ini'.server.domain=$OPENSHIFT_ROUTE_DOMAIN \
-  --set 'grafana\.ini'.server.root_url=https://v4m-grafana-$MON_NS.$OPENSHIFT_ROUTE_DOMAIN$OPENSHIFT_ROUTE_PATH_GRAFANA \
-  --set 'grafana\.ini'.server.serve_from_sub_path=$grafanaSubPath \
-  $versionstring \
-  --atomic \
-  $grafanaPwd \
-  $extraArgs \
-  v4m-grafana \
-  $chart2install
-
-if [ "$OPENSHIFT_AUTH_ENABLE" == "true" ]; then
-  log_info "Using OpenShift authentication for Grafana"
-  log_debug "Annotating grafana-serviceaccount to auto-generate TLS certs..."
-  kubectl annotate serviceaccount -n $MON_NS --overwrite grafana-serviceaccount 'serviceaccounts.openshift.io/oauth-redirectreference.primary={"kind":"OAuthRedirectReference","apiVersion":"v1","reference":{"kind":"Route","name":"v4m-grafana"}}'
-
-  log_debug "Adding ClusterRoleBinding for grafana-serviceaccount..."
-  crbYAML=$TMP_DIR/grafana-serviceaccount-binding.yaml
-  cp monitoring/openshift/grafana-serviceaccount-binding.yaml $crbYAML
-  if echo "$OSTYPE" | grep 'darwin' > /dev/null 2>&1; then
-    sed -i '' "s/__MON_NS__/$MON_NS/g" $crbYAML
-  else
-    sed -i "s/__MON_NS__/$MON_NS/g" $crbYAML
-  fi
-
-  kubectl apply -f $crbYAML
-
-  kubectl apply -n $MON_NS -f monitoring/openshift/grafana-proxy-secret.yaml
-  
-  grafanaProxyPatchYAML=$TMP_DIR/grafana-proxy-patch.yaml
-
-  if [ "$OPENSHIFT_PATH_ROUTES" == "true" ]; then
-    log_debug "Using path-based version of the OpenShift Grafana proxy patch"
+    -n "$MON_NS" \
+    -f "$imageKeysFile" \
+    -f "$wnpValuesFile" \
+    -f "$grafanaYAML" \
+    -f "$grafanaAuthYAML" \
+    -f "$autogeneratedYAMLFile" \
+    -f "$userGrafanaYAML" \
+    -f "$tempoDSFile" \
+    --set 'grafana\.ini'.server.domain="$OPENSHIFT_ROUTE_DOMAIN" \
+    --set 'grafana\.ini'.server.root_url="https://v4m-grafana-$MON_NS.$OPENSHIFT_ROUTE_DOMAIN$OPENSHIFT_ROUTE_PATH_GRAFANA" \
+    --set 'grafana\.ini'.server.serve_from_sub_path="$grafanaSubPath" \
+    $versionstring \
+    --atomic \
+    "${grafanaPwd:-}" \
+    "${extraArgs:-}" \
+    v4m-grafana \
+    "$chart2install"
+
+if [ "$OPENSHIFT_AUTH_ENABLE" = "true" ]; then
+    log_info "Using OpenShift authentication for Grafana"
+    log_debug "Annotating grafana-serviceaccount to auto-generate TLS certs..."
+    kubectl annotate serviceaccount -n "$MON_NS" --overwrite grafana-serviceaccount 'serviceaccounts.openshift.io/oauth-redirectreference.primary={"kind":"OAuthRedirectReference","apiVersion":"v1","reference":{"kind":"Route","name":"v4m-grafana"}}'
+
+    log_debug "Adding ClusterRoleBinding for grafana-serviceaccount..."
+    crbYAML=$TMP_DIR/grafana-serviceaccount-binding.yaml
+    cp monitoring/openshift/grafana-serviceaccount-binding.yaml "$crbYAML"
+    if echo "$OSTYPE" | grep 'darwin' > /dev/null 2>&1; then
+        sed -i '' "s/__MON_NS__/$MON_NS/g" "$crbYAML"
+    else
+        sed -i "s/__MON_NS__/$MON_NS/g" "$crbYAML"
```

Comment:
Replace lines 230-234 with call to the v4m_replace function: i.e. `v4m_replace "__MON_NS__" "$MON_NS" "$crbYAML"`?